src/mlpack/methods/cf/cf_main.cpp

```diff
@@ -364,16 +375,20 @@ void PerformAction(CFModel* c)
   CLI::GetParam<CFModel*>("output_model") = c;
 }
 
-template<typename DecompositionPolicy>
+template <typename DecompositionPolicy>
 void PerformAction(arma::mat& dataset,
                    const size_t rank,
                    const size_t maxIterations,
                    const double minResidue)
 {
   const size_t neighborhood = (size_t) CLI::GetParam<int>("neighborhood");
   CFModel* c = new CFModel();
+
+  const string normalizationType = CLI::GetParam<string>("normalization");
+
   c->template Train<DecompositionPolicy>(dataset, neighborhood, rank,
-      maxIterations, minResidue, CLI::HasParam("iteration_only_termination"));
+      maxIterations, minResidue, CLI::HasParam("iteration_only_termination"),
+      normalizationType);
```

Comment:
Don't feel obligated to change that, but I guess since `normalizationType` is only used once, we could inline it here as well, just like it was done for ` CLI::HasParam("iteration_only_termination")`.

Reply:
`normalizationType` is a string, I have used it to initialize the `CFType` object accordingly here: https://github.com/mlpack/mlpack/pull/2136/files#r362995086

Reply:
I see, works as well.